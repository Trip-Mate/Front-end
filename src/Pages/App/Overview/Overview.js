import React from 'react'
import { HomeRoute, NewTripRoute, ContactRoute } from '../../../Routing'
import { NavLink } from "react-router-dom";

// Material UI
import { makeStyles } from '@material-ui/core/styles';
import {
  Container,
  Typography,
}  from '@material-ui/core';

// Icons 
import FlightTakeoffIcon from '@material-ui/icons/FlightTakeoff';
import PersonIcon from '@material-ui/icons/Person';
import LanguageIcon from '@material-ui/icons/Language';
import AccessTimeIcon from '@material-ui/icons/AccessTime';
import AssignmentIndIcon from '@material-ui/icons/AssignmentInd';
import ShareIcon from '@material-ui/icons/Share';

const useStyles = makeStyles((theme) => ({
  Container: {
    marginTop: theme.spacing(8),
    margin: 'auto',
    display: 'grid',
    gridAutoFlow: 'column',
    
  },
  mainDiv: {
     display: 'grid',
  },

  icons: {
    gridAutoFlow: 'column',
    margin: 'auto ',
    border: '3px solid 	#3b5998',
    borderRadius: '50%' ,
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
    padding:'13px',
    fontSize: 'large',
    lineHeight: '0',
  },
  title:{
    color: '	#3b5998',
    alignItems: 'center',
    textAlign: 'center',
    fontSize: '18px',
    fontWeight: 'bolder',
    letterSpacing: '1px',
    BoxShadow: '5px',
    
  },
  image: {
    position: 'absolute',
    zIndex: '-1',
    top: '56px',
    backgroundSize: 'cover',
    width: '100%',
    height: '94vh',
    // filter: 'grayscale(100%)',
    opacity: '0.4',
    
  },
  link: {
    color: '	#3b5998 ',
  },
  hover: {
    '&:hover': {
      animation: `$spin 3s`,
   }
  },
    '@keyframes spin': {
      '25%': {
        transform:' rotateY(-60deg)',
        color: 'blue',
        animationTimingFunction: 'ease',
      },
      
      '100%': {
        color: 'blue',
        transform: 'rotateY(360deg)',
      },
    },
  
  
    // "&:hover": {
    //   backgroundColor: 'rgb(7, 177, 77, 0.42)'
    // }
  

}));

function Overview(props) {


  

  const classes = useStyles();

  return (
    <div>
    <img className={classes.image} 
    src='https://images.unsplash.com/photo-1527422265102-22027ee90fcd?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=80' 
    alt="background" />
    <Container className={classes.Container} >
          <div  className={classes.mainDiv}>
            <div  className={classes.icons}  >
                <NavLink className={classes.link}   to={HomeRoute} > 
                  <FlightTakeoffIcon   fontSize='large' className={classes.hover} />
                </NavLink>
              </div>
              <div>
                <Typography className={classes.title}>My Trips</Typography>
            </div>
          </div>
          <div  className={classes.mainDiv}>
            <div  className={classes.icons}  >
                <NavLink className={classes.link}   to={HomeRoute} > 
                  <PersonIcon   fontSize='large' className={classes.hover} />
                </NavLink>
              </div>
              <div>
                <Typography className={classes.title}>Home</Typography>
            </div>
          </div>
          <div  className={classes.mainDiv}>
            <div  className={classes.icons}  >
                <NavLink className={classes.link}   to={NewTripRoute} > 
                  <LanguageIcon   fontSize='large' className={classes.hover} />
                </NavLink>
              </div>
              <div>
                <Typography className={classes.title}>Plan Trip</Typography>
            </div>
          </div>
    </Container>

    <Container className={classes.Container} >
    <div  className={classes.mainDiv}>
            <div  className={classes.icons}  >
                <NavLink className={classes.link}   to={HomeRoute} > 
                  <AccessTimeIcon   fontSize='large' className={classes.hover} />
                </NavLink>
              </div>
              <div>
                <Typography className={classes.title}>Random?</Typography>
            </div>
          </div>
          <div  className={classes.mainDiv}>
            <div  className={classes.icons}  >
                <NavLink className={classes.link}   to={HomeRoute} > 
                  <AssignmentIndIcon   fontSize='large' className={classes.hover} />
                </NavLink>
              </div>
              <div>
                <Typography className={classes.title}>Contact Us</Typography>
            </div>
          </div><div  className={classes.mainDiv}>
            <div  className={classes.icons}  >
                <NavLink className={classes.link}   to={ContactRoute} > 
                  <ShareIcon   fontSize='large' className={classes.hover} />
                </NavLink>
              </div>
              <div>
                <Typography className={classes.title}>Share Trips</Typography>
            </div>
          </div>
    </Container>


  </div>
  )
}

export default Overview
