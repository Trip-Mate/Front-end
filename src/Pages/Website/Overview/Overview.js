import React from 'react';

function Overview() {
  return <></>;
}

export default Overview;

//TODO: new trip, current trip, All trips
