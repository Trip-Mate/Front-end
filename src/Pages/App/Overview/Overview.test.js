import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";

import Overview from "./Overview";

test("Overview snapshot matches", () => {
  const wrapper = shallow(<Overview />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});