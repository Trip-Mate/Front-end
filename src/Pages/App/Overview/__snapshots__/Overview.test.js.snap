// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview snapshot matches 1`] = `
<div>
  <img
    alt="background"
    className="makeStyles-image-5"
    src="https://images.unsplash.com/photo-1527422265102-22027ee90fcd?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=80"
  />
  <WithStyles(ForwardRef(Container))
    className="makeStyles-Container-1"
  >
    <div
      className="makeStyles-mainDiv-2"
    >
      <div
        className="makeStyles-icons-3"
      >
        <NavLink
          className="makeStyles-link-6"
          to="/trips/my-trips"
        >
          <FlightTakeoffIcon
            className="makeStyles-hover-7"
            fontSize="large"
          />
        </NavLink>
      </div>
      <div>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-4"
        >
          My Trips
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
    <div
      className="makeStyles-mainDiv-2"
    >
      <div
        className="makeStyles-icons-3"
      >
        <NavLink
          className="makeStyles-link-6"
          to="/"
        >
          <PersonIcon
            className="makeStyles-hover-7"
            fontSize="large"
          />
        </NavLink>
      </div>
      <div>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-4"
        >
          Home
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
    <div
      className="makeStyles-mainDiv-2"
    >
      <div
        className="makeStyles-icons-3"
      >
        <NavLink
          className="makeStyles-link-6"
          to="/trips/new-trip"
        >
          <LanguageIcon
            className="makeStyles-hover-7"
            fontSize="large"
          />
        </NavLink>
      </div>
      <div>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-4"
        >
          Plan Trip
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
  </WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-Container-1"
  >
    <div
      className="makeStyles-mainDiv-2"
    >
      <div
        className="makeStyles-icons-3"
      >
        <NavLink
          className="makeStyles-link-6"
          to="/"
        >
          <AccessTimeIcon
            className="makeStyles-hover-7"
            fontSize="large"
          />
        </NavLink>
      </div>
      <div>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-4"
        >
          Random?
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
    <div
      className="makeStyles-mainDiv-2"
    >
      <div
        className="makeStyles-icons-3"
      >
        <NavLink
          className="makeStyles-link-6"
          to="/"
        >
          <AssignmentIndIcon
            className="makeStyles-hover-7"
            fontSize="large"
          />
        </NavLink>
      </div>
      <div>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-4"
        >
          Contact Us
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
    <div
      className="makeStyles-mainDiv-2"
    >
      <div
        className="makeStyles-icons-3"
      >
        <NavLink
          className="makeStyles-link-6"
          to="/contact"
        >
          <ShareIcon
            className="makeStyles-hover-7"
            fontSize="large"
          />
        </NavLink>
      </div>
      <div>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-4"
        >
          Share Trips
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
  </WithStyles(ForwardRef(Container))>
</div>
`;
